packages/rollup/src/rollup_bundle.bzl

```diff
@@ -0,0 +1,257 @@
+
+Passed to the [`--input` option](https://github.com/rollup/rollup/blob/master/docs/999-big-list-of-options.md#input) in Rollup.
+
+Keys in this dictionary are labels pointing to .js entry point files.
+Values are the name to be given to the corresponding output chunk.
+
+Either this attribute or `entry_point` must be specified, but not both.
+""",
+        allow_files = [".js"],
+    ),
+    "format": attr.string(
+        doc = """"Specifies the format of the generated bundle. One of the following:
+
+- `amd`: Asynchronous Module Definition, used with module loaders like RequireJS
+- `cjs`: CommonJS, suitable for Node and other bundlers
+- `esm`: Keep the bundle as an ES module file, suitable for other bundlers and inclusion as a `<script type=module>` tag in modern browsers
+- `iife`: A self-executing function, suitable for inclusion as a `<script>` tag. (If you want to create a bundle for your application, you probably want to use this.)
+- `umd`: Universal Module Definition, works as amd, cjs and iife all in one
+- `system`: Native format of the SystemJS loader
+""",
+        values = ["amd", "cjs", "esm", "iife", "umd", "system"],
+        default = "esm",
+    ),
+    "globals": attr.string_dict(
+        doc = """Specifies id: variableName pairs necessary for external imports in umd/iife bundles.
+
+Passed to the [`--globals` option](https://github.com/rollup/rollup/blob/master/docs/999-big-list-of-options.md#outputglobals) in Rollup.
+Also, the keys from the map are passed to the [`--external` option](https://github.com/rollup/rollup/blob/master/docs/999-big-list-of-options.md#external).
+""",
+    ),
+    "output_dir": attr.string(
+        doc = """The directory in which all generated chunks are placed.
+
+By default, the directory is named the same as the target name.
+""",
+    ),
+    "rollup_bin": attr.label(
+        doc = "Target that executes the rollup binary",
+        executable = True,
+        cfg = "host",
+        default = "@npm//rollup/bin:rollup",
+    ),
+    "sourcemap": attr.bool(
+        doc = """Whether to produce a .js.map output
+
+Passed to the [`--sourcemap` option](https://github.com/rollup/rollup/blob/master/docs/999-big-list-of-options.md#outputsourcemap") in Rollup
+""",
+        default = True,
+    ),
+    "deps": attr.label_list(
+        aspects = [module_mappings_aspect],
+        doc = """Other libraries that are required by the code, or by the rollup.config.js""",
+    ),
+}
+
+def _chunks_dir_out(output_dir, name):
+    return output_dir if output_dir else name
+
+def _rollup_outs(sourcemap, name, entry_point, entry_points, output_dir):
+    """Supply some labelled outputs in the common case of a single entry point"""
+    result = {}
+    if entry_point:
+        # If the entry point ends with .js, we assume it is the primary entry point
+        result["entry_point_chunk"] = "/".join([_chunks_dir_out(output_dir, name), entry_point.name])
```

Comment:
It may be unexpected that switching from entry_point to entry_points with a single entry point has different named output behavior.

With the docstring `This is just a shortcut for the `entry_points` attribute with a single output chunk named the same as the input file.` I think the user would expect the named output to be `_no_ext(basename(entry_point))`?

Reply:
Fixed to share common `_desugar_entry_points{_names}` functions, and outside the desugaring we only read from `entry_points` to ensure they are equivalent

also added more docstring for this